### PR TITLE
Bring in KW fixes from master

### PR DESCRIPTION
**Pull request by nakulkorde**

### CODE REVIEW

**Comment by nakulkorde** [comment]
Docker build is failing. It appears that it is still referring to 1.4.0 somehow.

**Comment by tswhison** [comment]
> Docker build is failing. It appears that it is still referring to 1.4.0 somehow.

Yeah, I seem to remember a hard-coded reference to 1.4.0.  @r-rojo, could you have a look?

File: opae-libs/plugins/xfpga/metrics/metrics_utils.c [hunk 1]
```diff
@@ -128,17 +128,28 @@ fpga_result add_metric_vector(fpga_metric_vector *vector,
 	}
 
 	len = strnlen(group_name, SYSFS_PATH_MAX - 1);
-	strncpy(fpga_enum_metric->group_name, group_name, len + 1);
+	memcpy(fpga_enum_metric->group_name, group_name, len);
+	fpga_enum_metric->group_name[len] = '\0';
+
 	len = strnlen(group_sysfs, SYSFS_PATH_MAX - 1);
-	strncpy(fpga_enum_metric->group_sysfs, group_sysfs, len + 1);
+	memcpy(fpga_enum_metric->group_sysfs, group_sysfs, len + 1);
```
Comment (HananiaAmir) [inline opae-libs/plugins/xfpga/metrics/metrics_utils.c:135]: id the "len + 1" should be only len ?

Comment (tswhison) [reply opae-libs/plugins/xfpga/metrics/metrics_utils.c:135]: Good catch, yes it should be len.

**Pull request merged**